tests/unit/test-controller.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Test_Controller class file.
+ *
+ * @package wp_theme_migrator
+ */
+
+namespace Alley\WP\Theme_Migrator\Tests;
+
+use Alley\WP\Theme_Migrator\Context;
+use Alley\WP\Theme_Migrator\Controller;
+use ReflectionClass;
+
+/**
+ * Visit {@see https://phpunit.de/documentation.html} to learn more.
+ */
+class Test_Controller extends Test_Case {
+	/**
+	 * Set up.
+	 */
+	public function setup(): void {
+		$this->valid_context = new Context(
+			[
+				'theme'     => 'classic-theme',
+				'callbacks' => [ '__return_true' ],
+			]
+		);
+
+		parent::setup();
+	}
+	/**
+	 * Tests the functionality of the set_up_wp method.
+	 *
+	 * @throws ReflectionException Thrown if the class to reflect does not exist.
+	 */
+	public function test_set_up_wp() {
+		$controller = new Controller( $this->valid_context );
+
+		$class  = new ReflectionClass( 'Alley\WP\Theme_Migrator\Controller' );
+		$method = $class->getMethod( 'set_up_wp' );
+		$method->setAccessible( true );
+
+		$property = $class->getProperty( 'wp' );
+		$property->setAccessible( true );
+
+		$this->go_to( home_url( '/asdf/' ) );
+		$method->invoke( $controller );
+
+		$this->assertNotEmpty( $property->getValue( $controller )->query_vars );
+	}
+
+	/**
+	 * Data provider for test_set_migratability_during_cron method.
+	 *
+	 * @return array Array of data.
+	 */
+	public function data_set_migratability_during_cron(): array {
+		return [
+			'skip during cron'        => [
+				'__return_true',
+				false,
+			],
+			'don\'t skip during cron' => [
+				'__return_false',
+				true,
+			],
+		];
+	}
+
+	/**
+	 * Tests the functionality of the set_migratability method while DOING_CRON.
+	 *
+	 * @dataProvider data_set_migratability_during_cron
+	 *
+	 * @param callable $original The filter callback to test.
+	 * @param bool     $expected The expected result.
+	 *
+	 * @throws ReflectionException Thrown if the class to reflect does not exist.
+	 */
+	public function test_set_migratability_during_cron( callable $original, bool $expected ) {
+		if ( ! defined( 'DOING_CRON' ) ) {
```

Comment:
imo the `defined` check should be dropped - the test only has value if it is set to true, so if it's already defined when this test function runs and it's set to false, this test will proceed, when really it should error out because that value was already set somewhere else before this function ran.

Reply:
Interesting. As soon as I took out the conditional, it errored out. I'll have to look into why it's carrying over.